app/src/main/res/layout/activity_filtering_books.xml

```diff
@@ -0,0 +1,242 @@
+                        android:layout_height="wrap_content"
+                        android:orientation="horizontal" >
+
+                        <RadioButton
+                            android:id="@+id/popularity_sort"
+                            android:text="@string/popularity"
+                            android:layout_width="wrap_content"
+                            android:layout_height="match_parent" />
+
+                        <RadioButton
+                            android:id="@+id/title_inc_sort"
+                            android:text="@string/title_inc"
+                            android:layout_width="wrap_content"
+                            android:layout_height="match_parent" />
+
+                        <RadioButton
+                            android:id="@+id/title_dec_sort"
+                            android:layout_width="wrap_content"
+                            android:layout_height="match_parent"
+                            android:text="@string/title_dec" />
+
+                    </RadioGroup>
+                </HorizontalScrollView>
+            </LinearLayout>
+
+
+            <!-- Field parameter (eg. Computer Science, Biology) -->
+            <LinearLayout
+                android:id="@+id/field_view"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:orientation="vertical">
+
+                <TextView
+                    android:text="@string/field"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:textAllCaps="false"
+                    android:textSize="20sp" />
+
+                <HorizontalScrollView
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content" >
+
+                    <LinearLayout
+                        android:id="@+id/fieldGroup"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:orientation="horizontal">
+
+                        <CheckBox
+                            android:id="@+id/CS"
+                            android:text="@string/cs"
+                            android:layout_width="wrap_content"
+                            android:layout_height="match_parent" />
+
+                        <CheckBox
+                            android:id="@+id/Biology"
+                            android:text="@string/bio"
+                            android:layout_width="wrap_content"
+                            android:layout_height="match_parent" />
+
+                        <CheckBox
+                            android:id="@+id/Archi"
+                            android:text="@string/archi"
+                            android:layout_width="wrap_content"
+                            android:layout_height="match_parent" />
```

Comment:
I'm convinced those should be a Spinner, not CheckBoxes. There are way too many fields for it to be checkboxes.
But I'm aware Spinner are a bit trickier to code than CheckBoxes...